extensions/amp-consent/0.1/amp-consent.js

```diff
@@ -547,40 +534,98 @@ export class AmpConsent extends AMP.BaseElement {
   }
 
   /**
-   * Handle Prompt UI.
-   * @param {boolean} isConsentRequired
+   * @param {string} hasLocalStorageValue
    * @return {Promise<boolean>}
    */
-  initPromptUI_(isConsentRequired) {
-    this.consentUI_ = new ConsentUI(
-      this,
-      /** @type {!JsonObject} */ (devAssert(
-        this.consentConfig_,
-        'consent config not found'
-      ))
-    );
+  showPromptOrNot_(hasLocalStorageValue) {
+    // Remote case
+    if (this.consentConfig_['consentRequired'] === 'remote') {
+      userAssert(
+        this.consentConfig_['checkConsentHref'],
+        'checkConsentHref must be a valid endpoint, when using remote'
+      );
+      // If no local storage, then wait for this to tell us if consent is required
+      if (!hasLocalStorageValue) {
+        // Get remote consent using `checkConsentHref`
+        this.getConsentRemote_().then(consentInfo => {
+          const constentRequiredResponse = consentInfo['consentRequired'];
+          const consentStateValueResponse = getConsentItemStateForValue(
+            consentInfo['consentStateValue']
+          );
+
+          // check if consentRequired and no previous consent state stored
+          if (
+            constentRequiredResponse === true &&
+            consentStateValueResponse === CONSENT_ITEM_STATE.UNKNOWN
+          ) {
+            // Prompt (and eligible for postPromptUi)
+            this.scheduleDisplay_(false);
+            this.promptShown_ = true;
+          }
+          // last case is if consent is required but there `consentStateValueResponse` is not `unknown`
+          // then we don't do anything.
+        });
+      }
+    } else if (this.consentConfig_['consentRequired']) {
+      if (!hasLocalStorageValue) {
+        // Prompt (and eligible for postPromptUi)
+        this.scheduleDisplay_(false);
+        this.promptShown_ = true;
+        // TODO(@zhouyx):
+        // Race condition on consent state change between schedule to
+        // display and display. Add one more check before display
+      }
+    }
+  }
+
+  /**
+   * Attempts to sync with the server endpoint if there is one.
+   */
+  syncServerData_() {
+    // This is where passSharedData and clearCache will be as well as syncing with the cache
+    this.getConsentRemote_().then(response => {
+      // Pass shared data
```

Comment:
no need to over comment on the code. the method names are already explaining.
this applies to all the code